src/ILVerify/src/Program.cs

```diff
@@ -97,169 +85,146 @@ private ArgumentSyntax ParseCommandLine(string[] args)
-
-            try
+            ArgumentSyntax syntax = ParseCommandLine(args);
+            if (_help)
             {
-                var importer = new ILImporter(method, methodIL);
-
-                importer.ReportVerificationError = (args) =>
-                {
-                    var message = new StringBuilder();
-
-                    message.Append("[IL]: Error: ");
-
-                    message.Append("[");
-                    message.Append(_typeSystemContext.GetModulePath(((EcmaMethod)method).Module));
-                    message.Append(" : ");
-                    message.Append(((EcmaType)method.OwningType).Name);
-                    message.Append("::");
-                    message.Append(method.Name);
-                    message.Append("(");
-                    if (method.Signature._parameters != null && method.Signature._parameters.Length > 0)
-                    {
-                        foreach (TypeDesc parameter in method.Signature._parameters)
-                        {
-                            message.Append(parameter.ToString());
-                            message.Append(", ");
-                        }
-                        message.Remove(message.Length - 2, 2);
-                    }
-                    message.Append(")");
-                    message.Append("]");
+                Help(syntax.GetHelpText());
+                return 1;
+            }
 
-                    message.Append("[offset 0x");
-                    message.Append(args.Offset.ToString("X8"));
-                    message.Append("]");
+            if (_inputFilePaths.Count == 0)
+                throw new CommandLineException("No input files specified");
 
-                    if (args.Found != null)
-                    {
-                        message.Append("[found ");
-                        message.Append(args.Found);
-                        message.Append("]");
-                    }
+            _verifier = new Verifier(this);
+            _verifier.SetSystemModuleName(_systemModule);
 
-                    if (args.Expected != null)
-                    {
-                        message.Append("[expected ");
-                        message.Append(args.Expected);
-                        message.Append("]");
-                    }
+            foreach (var kvp in _inputFilePaths)
+            {
+                var results = VerifyAssembly(new AssemblyName(kvp.Key));
 
-                    if (args.Token != 0)
-                    {
-                        message.Append("[token  0x");
-                        message.Append(args.Token.ToString("X8"));
-                        message.Append("]");
-                    }
+                foreach (var result in results)
+                {
+                    PrintResult(result, kvp.Value);
+                }
 
-                    message.Append(" ");
+                var numErrors = results.Count();
```

Comment:
I think since `VerifyAssembly` is implemented using `yield return`, this will verify the entire assembly a second time, just to get the error count. It might make sense to simply increase a counter in the above foreach instead.

Reply:
Definitely. Thanks!